defaults/os_config_ref.yml

```diff
@@ -7,6 +7,7 @@ debian:
     # thoses 2 are globals for debian family:
     nextcloud_websrv_user: "www-data"
     nextcloud_websrv_group: "www-data"
+    upgrade_packages_first: false
```

Comment:
That's more of a decision how defaults are handled in this repo in general. I haven't added this in the first place, but then I was requested to add and I thought this default does not make sense to add to defaults/main.yml, because there are only defaults as to nextcloud so I thought it is more consistent in here. A default can always be overwritten by the user by adding it into the role. Putting it to false keeps the prior behaviour.

Reply:
> That's more of a decision how defaults are handled in this repo in general. 

I wish I had this much influence on the ansible project. 😉 but that's how default variables are declared for roles in ansible itself.  This is an ansible standard.  The other files are custom integration of facts to statistically change some variables without using tasks and set_facts module. I've put them in the defaults folder as they are reference facts _from my point of vue_. But they are only internally used from ansible perspective.

> I haven't added this in the first place, but then I was requested to add and I thought this default does not make sense to add to defaults/main.yml, because there are only defaults as to nextcloud so I thought it is more consistent in here.

If the issue is with the name or the placement  of the variable in the file, that's  Allright 😃 rename it if you wish, add a new section to the file, you have choices.

> A default can always be overwritten by the user by adding it into the role. Putting it to false keeps the prior.

Variables in ansible have a complex overriding system based by location in files and playbook code. Role's defaults  (in defaults/main.yml) have the lowest priority. That allows user to specify the same values somewhere  else. By that I mean that users don't  have to touch the role code at all, even the defaults/main.yml file

Reply:
Done.